packages/bus/src/brokers/RMQEventBus.ts

```diff
@@ -0,0 +1,479 @@
+    @inject(Configuration) private readonly sdkConfig: Configuration,
+    @inject(EventBusConfig) private readonly busConfig: EventBusConfig,
+    @inject(RetryStrategy)
+    private readonly backOffRetry: ExponentialBackoffRetryStrategy,
+    @inject(ConnectionFactory)
+    private readonly connectionFactory: ConnectionFactory<Connection>
+  ) {
+    this.container = sdkConfig.container;
+    this.subject.setMaxListeners(Infinity);
+  }
+
+  public async init(): Promise<void> {
+    const timeout = this.busConfig.connectTimeout;
+    const heartbeat = this.busConfig.heartbeatInterval;
+    const credentials = this.sdkConfig.credentials;
+    const url = this.sdkConfig.bus;
+
+    const buildedUrl = this.buildUrl({
+      url,
+      credentials,
+      heartbeat: heartbeat ?? 5000
+    });
+
+    this.client = await this.backOffRetry.acquire(() =>
+      this.connectionFactory.create(buildedUrl, {
+        timeout
+      })
+    );
+
+    this.client.on('close', (reason?: Error) =>
+      reason ? this.reconnect() : undefined
+    );
+
+    await this.createConsumerChannel();
+  }
+
+  public async register<T, R>(
+    type: EventHandlerConstructor<T, R>
+  ): Promise<void> {
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    const handlerInstance = this.getHandlerInstance(type);
+
+    const eventNames: string[] = this.reflectEventsNames(type, handlerInstance);
+
+    await Promise.all(
+      eventNames.map(eventName =>
+        this.registerEventHandler(handlerInstance, eventName)
+      )
+    );
+  }
+
+  public async unregister<T, R>(
+    type: EventHandlerConstructor<T, R>
+  ): Promise<void> {
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    const handlerInstance = this.getHandlerInstance(type);
+
+    const eventNames: string[] = this.reflectEventsNames(type, handlerInstance);
+
+    await Promise.all(
+      eventNames.map(eventName =>
+        this.unregisterEventHandler(handlerInstance, eventName)
+      )
+    );
+  }
+
+  public async publish<T>(event: Event<T>): Promise<void> {
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    const { type, payload, correlationId, createdAt } = event;
+
+    await this.backOffRetry.acquire(() =>
+      this.channel?.publish(
+        this.busConfig.exchange,
+        type,
+        Buffer.from(JSON.stringify(payload)),
+        {
+          type,
+          correlationId,
+          contentType: 'application/json',
+          mandatory: true,
+          persistent: true,
+          replyTo: this.REPLY_QUEUE_NAME,
+          timestamp: createdAt.getTime()
+        }
+      )
+    );
+  }
+
+  public async execute<T, R>(command: Command<T, R>): Promise<R> {
+    const { correlationId, ttl, expectReply } = command;
+
+    let waiter = Promise.resolve<unknown>(undefined);
+
+    if (expectReply) {
+      process.nextTick(() => {
+        waiter = this.expectReply<R>(correlationId, ttl);
+      });
+    }
```

Comment:
It doesn't work this way. `process.nextTick` should allow you to send a command right after creating a promise